src/Type/UnionType.php

```diff
@@ -245,7 +245,16 @@ public function isSuperTypeOfWithReason(Type $otherType): IsSuperTypeOfResult
 			return $otherType->isSubTypeOfWithReason($this);
 		}
 
-		$result = IsSuperTypeOfResult::createNo()->or(...array_map(static fn (Type $innerType) => $innerType->isSuperTypeOfWithReason($otherType), $this->types));
+		$results = [];
+		foreach ($this->types as $innerType) {
+			$result = $innerType->isSuperTypeOfWithReason($otherType);
+			if ($result->yes()) {
+				return $result;
+			}
+			$results[] = $result;
+		}
+
+		$result = IsSuperTypeOfResult::createNo()->or(...$results);
 		if ($result->yes()) {
```

Comment:
It cannot be a yes after this, righr. It could be deleted.

Reply:
right